src/v/datalake/tests/record_multiplexer_bench.cc

```diff
@@ -0,0 +1,564 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "container/fragmented_vector.h"
+#include "datalake/catalog_schema_manager.h"
+#include "datalake/record_multiplexer.h"
+#include "datalake/record_schema_resolver.h"
+#include "datalake/record_translator.h"
+#include "datalake/tests/catalog_and_registry_fixture.h"
+#include "datalake/tests/record_generator.h"
+#include "datalake/tests/test_data_writer.h"
+#include "model/record.h"
+#include "model/record_batch_reader.h"
+#include "serde/avro/tests/data_generator.h"
+#include "serde/protobuf/tests/data_generator.h"
+
+#include <seastar/testing/perf_tests.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/test/tools/old/interface.hpp>
+
+#include <optional>
+#include <string_view>
+
+namespace {
+std::string generate_nested_proto_int(size_t total_depth) {
+    constexpr auto proto_template = R"(
+    message Foo{} {{
+        {} 
+        string a{} = {}; 
+        {}
+    }})";
+
+    if (total_depth == 0) {
+        return "";
+    }
+    std::string member = "";
+    if (total_depth > 1) {
+        member = std::format(
+          "Foo{} b{} = {}; ",
+          total_depth - 1,
+          total_depth,
+          2 * total_depth + 1);
+    }
+    return std::format(
+      proto_template,
+      total_depth,
+      generate_nested_proto_int(total_depth - 1),
+      total_depth,
+      2 * total_depth,
+      member);
+}
+
+std::string generate_nested_proto(size_t total_depth) {
+    return std::format(
+      "syntax = \"proto3\"; {}", generate_nested_proto_int(total_depth));
+}
+
+std::string generate_linear_proto(size_t total_fields) {
+    constexpr auto proto_template = R"(
+    syntax = "proto3";
+    message Linear {{
+        {}
+    }})";
+
+    std::string fields = "";
+    for (int i = 1; i <= total_fields; i++) {
+        fields += std::format("string a{} = {};\n", i, i);
+    }
+
+    return std::format(proto_template, fields);
+}
+
+std::string generate_nested_avro_int(size_t total_depth) {
```

Comment:
nit: is "int" here "internal", like this is an internal call of a recursive method? Maybe spell out "internal" since it's easy to mistake this for having some kind of int field (same elsewhere)